src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -319,6 +323,23 @@ public RegisteredUser getUserFromSession(final HttpServletRequest request) {
             return null;
         }
 
+        if (properties.getProperty(Constants.SEGUE_APP_ENVIRONMENT).equals(EnvironmentType.PROD.name())) {
+            // If in production, check if the request originated from Isaac:
+            String referrer = request.getHeader("Referer");  // Note HTTP Header misspelling!
+            if (null == referrer) {
+                log.warn("Authenticated request had no 'Referer' information set!");
```

Comment:
Given that you are logging these to try and help find out in what circumstances they happen, would it be useful to log the request url seeing as you have the request object in scope?

Reply:
I think this is a good idea!

Reply:
Did you want the attempted to access part on this line too or just the ones below?

Reply:
It never happend :P
The magic of rebasing! Good spot again though, thanks for checking carefully!